contracts/levels/Switch.sol

```diff
@@ -0,0 +1,36 @@
+pragma solidity ^0.4.24;
+
+contract Switch {
+  bool public switchOn; // switch is off
+  bytes4 public offSelector = bytes4(keccak256("turnSwitchOff()"));
+
+  function flipSwitch(bytes _data) public
+      onlyOff
+      {
+      require(address(this).call(_data), "call failed :(");

```

Comment:
I wonder if there's a way to keep this as a self-referential external call without actually using `call()`, which immediately suggests "this is an unsafe spot". What about defining an interface and setting the contract address to `address(this)` in the constructor, then calling via the interface here?

Reply:
This is an interesting point. 
I had intentionally left the call there as a hint to being an `unsafe spot`. The previous levels I've made I've been careful not to make the problem too sneaky, as most of the levels were designed as educational. That is, to demonstrate a non-intuitive aspect of solidity that developers may not already know. 

Adding the interface definitely masks this, and adds slight complexity, which in my opinion may detract from the lesson this level is trying to teach. However I also agree with you that it will make it less obvious where the attack is. I'll make this change as a separate commit and let you decide what is best.

Reply:
On second thought, I'm not sure how you imagine this to work. Even with an interface, I think we still need to have a `CALL` somewhere. The dynamic type has to be given into a `CALL` in order for this trick to work, unless there's a function that does some assembly reading calldata etc. 

If you could be more explicit about how you imagine the interface to work, i'm happy to implement. 

Cheers